src/docs/guides/csp-nonce.md

```diff
@@ -0,0 +1,117 @@
+article walks through the process using Nginx and Webpack. Obviously, these resources don't account for the Stencil specifics, but we'll try to make
+those parts clear as we move on.
+
+Per the [MDN Guide on nonces](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/nonce#generating_values), a nonce should be "a random base64-encoded string of at least 128 bits of data from a cryptographically secure random number generator".
+
+### Output Targets
+
+Using nonces may differ slightly between output targets, so please be sure to use the correct pattern based on the context in which your
+Stencil components are consumed.
+
+#### Dist
+
+Consuming a `nonce` in the `dist` output target is easy using the provided `setNonce` helper function. This function is exported from the index
+file of the output target's designated output directory.
+
+This function simply accepts the `nonce` string value that you want set for every `style` and `script` tag.
+
+This is an example of consuming the `dist` output in an Angular app's entrypoint:
+
+```ts
+// main.ts
+
+import { defineCustomElements, setNonce } from 'my-lib/loader';
+
+// Will set the `nonce` attribute for all scripts/style tags
+// i.e will run styleTag.setAttribute('nonce', 'r4nd0m')
+// Obviously, you should use the nonce generated by your server
+setNonce('r4nd0m');
+
+// Generic Angular bootstrapping
+platformBrowserDynamic()
+  .bootstrapModule(AppModule)
+  .catch(err => console.log(err));
+
+defineCustomElements();
+```
+
+#### Custom Elements
+
+Consuming a `nonce` in the `dist-custom-elements` output target is easy using the provided `setNonce` helper function. This function is exported
+from the index file of the output target's designated output directory.
+
+This function simply accepts the `nonce` string value that you want set for every `style` and `script` tag.
+
+This is an example of consuming the `dist-custom-elements` output in an Angular app's entrypoint:
+
+```ts
+// main.ts
+
+import { defineCustomElements, setNonce } from 'my-lib/dist/components';
+// We'll assume we have `autoDefineCustomElements` enabled
+import 'my-lib/dist/components/my-component';
+
+// Will set the `nonce` attribute for all scripts/style tags
+// i.e will run styleTag.setAttribute('nonce', 'r4nd0m')
+// Obviously, you should use the nonce generated by your server
+setNonce('r4nd0m');
+
+// Generic Angular bootstrapping
+platformBrowserDynamic()
+  .bootstrapModule(AppModule)
+  .catch(err => console.log(err));
+```
+
+#### WWW
+
+Unfortunately, setting `nonce` attributes gets a bit trickier when it comes to [SSR and SSG](docs/static-site-generation). As a `nonce` needs
+to be unique per page view, we cannot define/set the nonce at build time. So, this responsibility now falls on the
+[hydrate app](docs/hydrate-app)'s execution of runtime code.
```

Comment:
IIRC, these links need a leading slash. ATM, this points to docs/docs/hydrate-app in the generated docs

```suggestion
[hydrate app](/docs/hydrate-app)'s execution of runtime code.
```